m4opt/models/background/skybright.py

```diff
@@ -0,0 +1,116 @@
+try:
+    from functools import cache
+except ImportError:  # FIXME: drop once we require Python >= 3.9
+    from functools import lru_cache as cache
+
+from importlib import resources
+import numpy as np
+from astropy.table import QTable
+import astropy.units as u
+from astropy.modeling.models import Tabular1D
+
+
+from . import data
+from .core import Background
+
+
+kpno_sky_tables = {'low': '10JunZen.txt', 'medium': '10FebZen.txt',
+                   'high': '10Phx.txt', 'veryhigh': '10Tuc.txt'}
+
+
+@cache
+def read_kpno_sky_data(option="medium"):
+    """
+    Data taken from Neugent & Massey(2010)[1]_; as per private communication
+    with P. Massey, sky brightness is given in units AB magnitudes/arcsec^2
+
+    Options include:
+    - "low"    : As measured at Zenith in June; lowest level of sky brightness
+    - "medium" : At Zenith in February
+    - "high"   : As measured at a zenith distance of 60 degrees towards Phoenix
+    - "veryhigh" : Measured at a zenith distance of 60 degrees towards Tucson
+
+    Notes
+    -----
+
+    Spectra is given only for B and V bands (~3750 - ~6875 Angstrom). This does
+    not yet include effect of lunar phase.
+
+    References
+    ----------
+    .. [1] Neugent, K. and Massey, P., 2010, "The Spectrum of the Night Sky
+           Over Kitt Peak: Changes Over Two Decades". PASP, 122, 1246-1253
+           doi:10.1086/656425
+    """
+
+    if option.lower() not in kpno_sky_tables.keys():
+        raise AttributeError("option must be one of {0}".format(
```

Comment:
Should be a `ValueError` ("Raised when an operation or function receives an argument that has the right type but an inappropriate value")